src/audio/pipeline/pipeline-schedule.c

```diff
@@ -60,30 +63,42 @@ static enum task_state pipeline_task(void *arg)
 		return SOF_TASK_STATE_RESCHEDULE;
 	}
 
-	if (p->trigger.cmd >= 0) {
+	if (cmd >= 0) {
 		/* First pipeline task run for either START or RELEASE: PRE stage */
-		struct sof_ipc_reply reply = {
-			.hdr.cmd = SOF_IPC_GLB_REPLY,
-			.hdr.size = sizeof(reply),
-		};
-		int cmd = p->trigger.cmd;
-
 		err = pipeline_trigger_run(p, p->trigger.host, cmd);
 		if (err < 0) {
 			pipe_err(p, "pipeline_task(): failed to trigger components: %d", err);
 			reply.error = err;
 			err = SOF_TASK_STATE_COMPLETED;
-		} else if (err == PPL_STATUS_PATH_STOP) {
-			pipe_warn(p, "pipeline_task(): stopping for xrun");
-			err = SOF_TASK_STATE_COMPLETED;
-		} else if (p->trigger.cmd != cmd) {
-			/* PRE stage completed */
-			if (p->trigger.delay)
-				return SOF_TASK_STATE_RESCHEDULE;
-			/* No delay: the final stage has already run too */
 		} else {
-			p->status = COMP_STATE_ACTIVE;
-			err = SOF_TASK_STATE_RESCHEDULE;
+			switch (cmd) {
```

Comment:
Can you state the new stop/pause flow in the commit message. I assume we dont have any race for sending reply IPCs that the stream has been stopped or paused.